### PR TITLE
JSHandle docs

### LINKED ISSUE #240: JSHandle Class XML Documentation

### PR DESCRIPTION
**Pull request by kblok**
closes #240

### CODE REVIEW

File: lib/PuppeteerSharp/JSHandle.cs [hunk 1]
```diff
@@ -7,23 +7,46 @@
...
+            Client = client;
+            Logger = Client.Connection.LoggerFactory.CreateLogger(this.GetType());
             RemoteObject = remoteObject;
         }
 
+        /// <summary>
+        /// Gets the execution context.
+        /// </summary>
+        /// <value>The execution context.</value>
         public ExecutionContext ExecutionContext => _context;
```
Comment (Meir017) [inline lib/PuppeteerSharp/JSHandle.cs]: this can be a get only property

Comment (kblok) [reply lib/PuppeteerSharp/JSHandle.cs]: It is a get only property 🤔

Comment (Meir017) [reply lib/PuppeteerSharp/JSHandle.cs]: we don't need the private field...

Comment (kblok) [reply lib/PuppeteerSharp/JSHandle.cs]: Ah yeah 😛

**Comment by kblok** [comment]
Done @Meir017. Now let's wait 2 days till all builds run 😝

File: lib/PuppeteerSharp/JSHandle.cs [hunk 2]
```diff
@@ -7,23 +7,44 @@
...
+            Logger = Client.Connection.LoggerFactory.CreateLogger(this.GetType());
             RemoteObject = remoteObject;
         }
 
-        public ExecutionContext ExecutionContext => _context;
+        /// <summary>
+        /// Gets the execution context.
+        /// </summary>
+        /// <value>The execution context.</value>
+        public ExecutionContext ExecutionContext { get; private set; }
```
Comment (Meir017) [inline lib/PuppeteerSharp/JSHandle.cs]: I don't think we need the `private set` you can set in the ctor even without it

Comment (kblok) [reply lib/PuppeteerSharp/JSHandle.cs]: cool, I didn't know about that.

Comment (kblok) [reply lib/PuppeteerSharp/JSHandle.cs]: Done @Meir017

File: lib/PuppeteerSharp.Tests/PuppeteerSharp.Tests.csproj [hunk 3]
```diff
@@ -18,6 +19,10 @@
   <PropertyGroup Condition="'$(TargetFramework)'=='net471'">
     <RuntimeIdentifier>win7-x64</RuntimeIdentifier>
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
```
Comment (Meir017) [inline lib/PuppeteerSharp.Tests/PuppeteerSharp.Tests.csproj]: what are these for?

File: lib/PuppeteerSharp/JSHandle.cs [hunk 4]
```diff
@@ -7,23 +7,44 @@
...
+        /// <summary>
+        /// Gets or sets a value indicating whether this <see cref="PuppeteerSharp.JSHandle"/> is disposed.
+        /// </summary>
+        /// <value><c>true</c> if disposed; otherwise, <c>false</c>.</value>
         public bool Disposed { get; set; }
+        /// <summary>
+        /// Gets or sets the remote object.
+        /// </summary>
+        /// <value>The remote object.</value>
         public dynamic RemoteObject { get; internal set; }
```
Comment (Meir017) [inline lib/PuppeteerSharp/JSHandle.cs]: do we need the `internal set` here?

File: lib/PuppeteerSharp.Tests/PuppeteerSharp.Tests.csproj (hunk 3 above)
Comment (kblok) [reply lib/PuppeteerSharp.Tests/PuppeteerSharp.Tests.csproj]: Damned. I have to revert this. My VS is hanging when I try to debug test. I'm trying to see what's going on. I accidentally committed this.

Comment (kblok) [reply lib/PuppeteerSharp.Tests/PuppeteerSharp.Tests.csproj]: Reverted.

File: lib/PuppeteerSharp/JSHandle.cs (hunk 4 above)
Comment (kblok) [reply lib/PuppeteerSharp/JSHandle.cs]: No, we don't. I removed that and I also change the Disposed setter to private.

**Pull request merged**